tools/dcos_login.py

```diff
@@ -203,16 +174,80 @@ def ignore_empty(envvar, default):
     dcos_login_username = ignore_empty("DCOS_LOGIN_USERNAME", __CLI_LOGIN_EE_USERNAME)
     dcos_login_password = ignore_empty("DCOS_LOGIN_PASSWORD", __CLI_LOGIN_EE_PASSWORD)
     dcos_enterprise = ignore_empty("DCOS_ENTERPRISE", "true").lower() == "true"
-    dcos_acs_token = os.environ.get("DCOS_ACS_TOKEN")
-    if not dcos_acs_token:
-        dcos_acs_token = login(
-            dcosurl=cluster_url,
-            username=dcos_login_username,
-            password=dcos_login_password,
-            is_enterprise=dcos_enterprise,
-        )
-    configure_cli(dcosurl=cluster_url, token=dcos_acs_token)
+    dcos_acs_token = os.environ.get("DCOS_ACS_TOKEN", login(
+        dcosurl=cluster_url,
+        username=dcos_login_username,
+        password=dcos_login_password,
+        is_enterprise=dcos_enterprise,
+    ))
+    if is_old_cli:
+        # TODO(takirala) : We can remove this code path few months after 1.12 GA
+        if dcos_enterprise:
+            _run_cmd("dcos auth login {} --username {} --password {}".format(
+                cluster_url,
+                dcos_login_username,
+                dcos_login_password))
+        else:
+            _run_cmd("dcos config set core.dcos_url {}".format(cluster_url))
+            _run_cmd("dcos config set core.ssl_verify {}".format(False))
+            _run_cmd("dcos config set core.dcos_acs_token {}".format(dcos_acs_token))
+    else:
+        if dcos_enterprise:
+            _run_cmd("dcos cluster setup {} --username {} --password {} --insecure".format(
+                cluster_url,
+                dcos_login_username,
+                dcos_login_password))
+        else:
+            dcos_acs_token = os.environ.get("DCOS_ACS_TOKEN", dcos_acs_token)
+            _run_cmd("env DCOS_ACS_TOKEN={} dcos cluster setup {} --insecure".format(
```

Comment:
It should also be possible to pass `__CLI_LOGIN_OPEN_TOKEN` to stdin in order to avoid retrieving the acs token manually. The only thing is that it'd open a browser when run on desktop environments.